integration/secret/secret_test.go

```diff
@@ -41,3 +43,110 @@ func TestSecretInspect(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, secret.ID, secretResp.ID)
 }
+
+func TestSecretList(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	defer setupTest(t)()
+	d := swarm.NewSwarm(t, testEnv)
+	defer d.Stop(t)
+	client, err := request.NewClientForHost(d.Sock())
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	testName0 := "test0"
+	testName1 := "test1"
+
+	// create secret test0
+	secret0, err := client.SecretCreate(ctx, swarmtypes.SecretSpec{
+		Annotations: swarmtypes.Annotations{
+			Name:   testName0,
+			Labels: map[string]string{"type": "test"},
+		},
+		Data: []byte("TESTINGDATA0"),
+	})
+	require.NoError(t, err)
+	assert.NotEqual(t, secret0.ID, "")
+
+	secret, _, err := client.SecretInspectWithRaw(ctx, secret0.ID)
+	require.NoError(t, err)
+	assert.Equal(t, secret.Spec.Name, testName0)
+
+	// create secret test1
+	secret1, err := client.SecretCreate(ctx, swarmtypes.SecretSpec{
+		Annotations: swarmtypes.Annotations{
+			Name:   testName1,
+			Labels: map[string]string{"type": "production"},
+		},
+		Data: []byte("TESTINGDATA1"),
+	})
+	require.NoError(t, err)
+	assert.NotEqual(t, secret1.ID, "")
+
+	secret, _, err = client.SecretInspectWithRaw(ctx, secret1.ID)
+	require.NoError(t, err)
+	assert.Equal(t, secret.Spec.Name, testName1)
+
+	names := func(entries []swarmtypes.Secret) []string {
+		values := []string{}
+		for _, entry := range entries {
+			values = append(values, entry.Spec.Name)
+		}
+		return values
+	}
+
+	// test by `secret ls`
+	entries, err := client.SecretList(ctx, types.SecretListOptions{})
+	require.NoError(t, err)
+	assert.Contains(t, names(entries), testName0)
+	assert.Contains(t, names(entries), testName1)
+
+	// test filter by name `secret ls --filter name=xxx`
+	f := filters.NewArgs()
+	f.Add("name", testName0)
+	entries, err = client.SecretList(ctx, types.SecretListOptions{
+		Filters: f,
```

Comment:
This can be done inline now instead of accross 3 lines:

```go
types.SecretListOptions{
     Filters: filters.NewArgs(filters.Arg("name", testName0)),
}
```

Reply:
@dnephin Done. Thanks.